src/adversarial/neurify.jl

```diff
@@ -126,27 +162,39 @@ function constraint_refinement(solver::Neurify, nnet::Network, reach::SymbolicIn
     return intervals
 end
 
-function get_nodewise_gradient(nnet::Network, LΛ::Vector{Vector{Float64}}, UΛ::Vector{Vector{Float64}})
+function get_nodewise_gradient(solver::Neurify, nnet::Network, LΛ::Vector{Vector{Float64}}, UΛ::Vector{Vector{Float64}})
     n_output = size(nnet.layers[end].weights, 1)
     n_length = length(nnet.layers)
-    LG = Matrix(1.0I, n_output, n_output)
-    UG = Matrix(1.0I, n_output, n_output)
+    LG = ones(n_output)
+    UG = ones(n_output)
     max_tuple = (0, 0, 0.0)
     for (k, layer) in enumerate(reverse(nnet.layers))
         i = n_length - k + 1
-        for j in size(layer.bias)
-            if LΛ[i][j] ∈ (0.0, 1.0) && UΛ[i][j] ∈ (0.0, 1.0)
-                max_gradient = max(abs(LG[j]), abs(UG[j]))
-                if max_gradient > max_tuple[3]
-                    max_tuple = (i, j, max_gradient)
+        if layer.activation != Id() 
+            # Not sure whether this is right, but by experiments, this caused infinity loop. 
```

Comment:
@changliuliu do you know the answer here? Also potentially with regards to other changes affecting correctness / logic flow.

Reply:
This comment is out-dated. What I was not sure is whether this could prevent an infinity loop. But later, I defined the `splitted` set, which can for sure prevent an infinity loop.
The logic here should be right, the intuition is only ReLU nodes should be split to reduce over-approximation.
I will change the comment.